istio/oam.go

```diff
@@ -11,46 +11,33 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+// CompHandler is the type for functions which can handle OAM components
+type CompHandler func(*Istio, v1alpha1.Component, bool) (string, error)
+
 // HandleComponents handles the processing of OAM components
 func (istio *Istio) HandleComponents(comps []v1alpha1.Component, isDel bool) (string, error) {
 	var errs []error
 	var msgs []string
-	for _, comp := range comps {
-		if comp.Spec.Type == "IstioMesh" {
-			if err := handleComponentIstioMesh(istio, comp, isDel); err != nil {
-				errs = append(errs, err)
-			}
-
-			msg := "created service of type \"IstioMesh\""
-			if isDel {
-				msg = "deleted service of type \"IstioMesh\""
-			}
-
-			msgs = append(msgs, msg)
-			continue
-		}
-
-		if comp.Spec.Type == "VirtualService" {
-			if err := handleComponentVirtualService(istio, comp, isDel); err != nil {
-				errs = append(errs, err)
-			}
 
-			msg := fmt.Sprintf("created virtual service \"%s\" in namespace \"%s\"", comp.Name, comp.Namespace)
-			if isDel {
-				msg = fmt.Sprintf("deleted virtual service \"%s\" in namespace \"%s\"", comp.Name, comp.Namespace)
-			}
+	compFuncMap := map[string]CompHandler{
+		"IstioMesh":            handleComponentIstioMesh,
```

Comment:
@utkarsh-pro Can this be moved to `config/`? Just the declarations.

Reply:
Won't that create circular dependency issue? This map's values are functions of type `func(*Istio, v1alpha1.Component, bool) (string, error)`, so it needs `Istio` from this package.

Reply:
makes sense. On a different note, are the contents inside `oam/` common across adapters? If not we can move this into that and import. Basically to manage this list independently in a file.

Reply:
Actually almost everything that was generic here has been already moved to `meshery-adapter-library` :sweat_smile: